### ISSUE TITLE
Replace triplea_maps.yaml with Github API

### ISSUE THREAD

**Issue by RoiEXLab** [triplea-game/triplea#1444]
I just had an Idea I need to share with you...
As you all know, the map list is generated using the triplea_maps file in this repository.
GitHub has a great HTTP-POST > JSON API which easily returns [every repository on the triplea-maps org](https://developer.github.com/v3/repos/#list-organization-repositories).
If every map repository had an INFO file describing itself, we could check that file to get every info of a map needed.
This way there's no longer a need for updating the triplea_maps file when adding a new map to the org.
If we wanted this file available offline, we could cache this request for times when internet is not available.
Also we could provide a "minimum version" attribute to the map repositories in order to prevent the client from updating to a map version made for a future version of triplea.

This also plays well with #1436, since the maps a required to ship an info file within themselves instead of the currently generated properties file

Please tell me what you think about this!

EDIT: [triplea-maps output](https://gist.github.com/RoiEXLab/d38ebbefb283e35b065111e7369f9bf7)

**Comment by DanVanAtta** [comment]
Interesting idea. Categories is the biggest item that comes to my mind. Managing them in a very distributed setting, each category setting found in each map repo, could potentially be challenging. Any thoughts on how categories would/could be managed?

I don't think we will want to require an http request per map repo to get the full set of data needed to show the download window. github does have rate limits which could cause unexplained lockouts. Though this API would be enough to populate the list, and we can then send background and/or on-click requests to get the map details as needed when users select map items in the maps download window.

We will also likely be including everything in the maps repo. We could maybe move the 'Project' repo out of the triplea-maps org and put it in with the triplea-game org, call it something like 'maps-project' . I don't think it's necessarily a bad thing otherwise to be including all maps.

**Comment by RoiEXLab** [comment]
@DanVanAtta 
I thought about this for a day, and AFAIK there's only one way to do this, via the repository description.
When using the Github API we could modify the repository description, which is part of the returned JSON and change it to the category name.
We probably want a proper formatted map name too, so we could either replace all underscores in repo names with spaces (also returned by JSON), or wwe format the description in a way it can hold both strings

**Comment by simon33-2** [comment]
So does this remove the need for updating the triplea_maps.yaml? Sounds great!

**Comment by RoiEXLab** [comment]
Yes, using this way, every map holds it's own information seperately.
If you update a Map, all you need to change is the repository itself

**Comment by ron-murhammer** [comment]
So cool idea though I think there are a few challenges:
1. Categorization (Dan already mentioned) - might be able to put it in the description but that could be difficult to manage
2. Update notification - Is there a way to know when a map has changed (new version)?

**Comment by simon33-2** [comment]
Is it possible to put something in the build.gradle file or whatever it is called to push updates to some place to flag map updates. I imagine permissions could be a problem with this.

**Comment by RoiEXLab** [comment]
@ron-murhammer The api provides a "last-pushed" timestamp, which we could compare against the timestamp of the saved map - we would need to save a last pushed timestamp of every map we downloaded though...

**Comment by RoiEXLab** [comment]
A new Feature was added to github called "Topics", basically a tag for every repository, which can be chosen relatively freely (numbers, letters an hyphens only, but that should be enough).
Unfortunately these topics are not accessible in the API yet, I contacted the Github support to get more information on that topic. But once they are, the are our solution to the categorization problem @ron-murhammer mentioned. We could add a topic like "tripleA-Good-Map" which is returned by the JSON then and validated by our client.
